app/src/main/java/com/babylon/wallet/android/presentation/transaction/submit/TransactionSubmitDelegate.kt

```diff
@@ -80,29 +84,34 @@ class TransactionSubmitDelegate(
                     message = failure.getDappMessage()
                 )
             }
-            onSendScreenEvent(Event.Dismiss)
+            _state.update {
+                it.copy(isTransactionDismissed = true)
+            }
```

Comment:
so the thing here is that I couldn't inject or with another way to pass the callback in the delegate.
If someone knows a way please let me know.

So I decided to go with the "Google" way by adding a new property in the viewmodel's state and observe it from the screen.

See my next comment.